bundles/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/RFXComHandlerFactory.java

```diff
@@ -46,25 +47,22 @@
 @NonNullByDefault
 @Component(service = ThingHandlerFactory.class, configurationPid = "binding.rfxcom")
 public class RFXComHandlerFactory extends BaseThingHandlerFactory {
-    /**
-     * Service registration map
-     */
-    private Map<ThingUID, ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
 
     private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Stream
             .concat(RFXComBindingConstants.SUPPORTED_DEVICE_THING_TYPES_UIDS.stream(),
                     RFXComBindingConstants.SUPPORTED_BRIDGE_THING_TYPES_UIDS.stream())
             .collect(Collectors.toSet());
 
-    private @NonNullByDefault({}) SerialPortManager serialPortManager;
+    /**
+     * Service registration map
+     */
+    private final Map<ThingUID, ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
 
-    @Reference
-    protected void setSerialPortManager(final SerialPortManager serialPortManager) {
-        this.serialPortManager = serialPortManager;
-    }
+    private final SerialPortManager serialPortManager;
 
-    protected void unsetSerialPortManager(final SerialPortManager serialPortManager) {
-        this.serialPortManager = null;
+    @Activate
```

Comment:
Afaik, "Activate" should only be added to an activate method, but not to the constructor.

Reply:
This is done like that everywhere I believe.

Reply:
For example: https://github.com/openhab/openhab-addons/blob/2.5.x/bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosHandlerFactory.java#L73

Reply:
Ok, seems a legitimate case: https://docs.osgi.org/javadoc/osgi.cmpn/7.0.0/org/osgi/service/component/annotations/Activate.html